src/main/scala/ai/privado/entrypoint/ScanProcessor.scala

```diff
@@ -328,6 +331,7 @@ object ScanProcessor extends CommandProcessor {
       case Failure(exception) => {
         logger.error("Error while parsing the source code!")
         logger.debug("Error : ", exception)
+        MetricHandler.setScanStatus(false)
         Left("Error while parsing the source code: " + exception.toString)
```

Comment:
Do we capture this error message outside of this function?
If not, we should capture the error too

Reply:
We capture the error outside of this function so it's not included here.

Reply:
@abhstabs Should not both metrics, for one thing, be handled together.. if that makes the code cleaner?

Reply:
The errors are logged one by one as they occur on a lower level. This is an umbrella exception to handle the user side messages.

Reply:
@abhstabs Makes sense. 
Just want to ensure that even "unhandled exceptions" that are raised from lower-level functions are caught, which might not be handled anywhere else but only in the top handler. Looks good if you can confirm that.

Reply:
Can confirm. Scan errors and upload errors are reported in the metrics at the lower level.